tests/test_reload.py

```diff
@@ -0,0 +1,43 @@
+import os
+from qgis.core import QgsProject
+import requests
+from xml.etree import ElementTree
+from shutil import copyfile
+import pytest
+
+
+@pytest.fixture(scope="session", autouse=True)
+def set_up():
+    # Will be executed before the first test
+    copyfile("project.qgs", "base_project.qgs")
+    yield
+    # Will be executed after the last test
+    copyfile("base_project.qgs", "project.qgs")
+    os.remove("base_project.qgs")
+
+
+def _get_capabilities():
+    url = "http://172.17.0.1:8380/?SERVICE=WMS&REQUEST=GetCapabilities"
+    response = requests.get(url)
+
+    tree = ElementTree.fromstring(response.content)
+    layers = tree.getchildren()[1].getchildren()[3]
+    layer_count = 0
+    for layer in layers.getchildren():
+        if "Layer" in layer.tag:
+            layer_count += 1
+    return layer_count
+
+
+def test_get_initial_layer_count():
+    assert _get_capabilities(), 2
+
+
+def test_remove_layer_from_project():
+
+    project = QgsProject.instance()
+    project.read("project.qgs")
+    id_layer = list(project.mapLayers().keys())[0]
+    project.removeMapLayer(id_layer)
+    project.write()
+    assert _get_capabilities(), 1
```

Comment:
test also passes without the plugin, ie if [line 11 in docker-compose](https://github.com/camptocamp/QgisServerProjectReloader/blob/644f93cbe82f61769431e7bc2c960fcd8f5607ef/docker-compose.yaml#L11) is commented

I did not manage to access the QGIS logs, so I could not check the plugin in work

Reply:
so maybe the current server setup already includes the reload function ? Is it a dev server ?

Reply:
Yes the current server should already include a server restart based on inotify.
The goal is to include this plugin in place to reload only the project, not the server, and to sopport database stored projects.
This is a circular dependency camptocamp/qgis-server will depends on this plugin whose tests depends on camptocamp/qgis-server. We could use official QGIS docker image here.